x-pack/plugins/cloud_integrations/cloud_data_migration/server/config.ts

```diff
@@ -9,7 +9,12 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import { PluginConfigDescriptor } from '@kbn/core-plugins-server';
 
 const configSchema = schema.object({
-  enabled: schema.boolean({ defaultValue: true }),
+  enabled: schema.conditional(
```

Comment:
Unfortunately, this breaks the common functional tests (I think related to work done via https://github.com/elastic/kibana/pull/159272). If I run the tests using the `common` config:

```
node scripts/functional_tests_server.js --config test_serverless/api_integration/test_suites/common/config.ts
```

Then, I get the following error:

```
 FATAL  Error: [config validation of [xpack.cloud_integrations.data_migration].enabled]: a value wasn't expected to be present
```

Reply:
Yeah while this approach makes things safer in terms of minimizing our API surface area, the tradeoff is that serverless & self-managed need to be tested separately & common tests won't really work anymore.

Reply:
This is actually related to the common serverless test suite. It looks like in https://github.com/elastic/kibana/pull/159272, a change was made so that the common suite runs directly with the `serverless.yml` file instead of specifying a project type.  See: https://github.com/elastic/kibana/pull/159272/files#diff-818fc0329318f82bcee0547dfb8edb3800cce36ea3fad7c62d58363197d91366R28

Reply:
Ah, I misunderstood your comment. Looking at the code I'm wondering if what's throwing it off is that we aren't getting a `--serverless` cli flag at all when running the common tests. Ultimately that's what we look at in the environment to set the context ref accordingly, so `schema.contextRef('serverless')` is likely returning `false`.

Poking around in `src/cli/serve.js` it appears we try to read from the provided config files as a backup method to determine if we're in serverless mode (if the `--serverless` flag isn't explicitly passed), however I think we are checking for a top-level `serverless` value to be in the yml files, which I don't think we provide anywhere... so we may have a bug in that logic. 

FYI @jbudz, this might be a bug, but keep me honest if I'm misreading things :)

Reply:
Sorry, should have clarified 😅 

Ah, I see. @jbudz let me know if you'd like me to open up an issue or if I can help in any way.

Reply:
Fixed via https://github.com/elastic/kibana/pull/160783